File: FootScout-Vue.WebAPI/Entities/ClubHistory.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace FootScout_Vue.WebAPI.Entities
{
    public class ClubHistory
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int PlayerPositionId { get; set; }

        [ForeignKey("PlayerPositionId")]
        public virtual PlayerPosition PlayerPosition { get; set; }

        [Required]
        [MaxLength(30)]
        public string ClubName { get; set; }

        [Required]
        [MaxLength(30)]
        public string League { get; set; }

        [Required]
        [MaxLength(30)]
        public string Region { get; set; }

        [Required]
        [DataType(DataType.DateTime)]
        public DateTime StartDate { get; set; }

        [Required]
        [DataType(DataType.DateTime)]
        public DateTime EndDate { get; set; }

        [Required]
        public int AchievementsId { get; set; }

        [ForeignKey("AchievementsId")]
        public virtual Achievements Achievements { get; set; }

        [Required]
        public string PlayerId { get; set; }

        [ForeignKey("PlayerId")]
        public virtual User Player { get; set; }
    }
}

File: FootScout-Vue.WebClient/src/models/interfaces/ClubHistory.ts
import type { Achievements } from '../dtos/AchievementsDTO';
import type { UserDTO } from '../dtos/UserDTO';
import type { PlayerPosition } from './PlayerPosition';

// Opisuje historię klubową zawodnika, zawierając informacje o klubie, lidze, pozycji na boisku i osiągnięciach.

export interface ClubHistoryModel {
    id: number;
    playerPositionId: number;
    playerPosition: PlayerPosition;
    clubName: string;
    league: string;
    region: string;
    startDate: string;
    endDate: string;
    achievementsId: number;
    achievements: Achievements;
    playerId: string;
    player: UserDTO;
}